### ISSUE TITLE
Can't use secret interpolation in irc services `password`

### ISSUE THREAD

**Issue by pmatos** [buildbot/buildbot#3668]
Related to #3627, we can't use password secret for IRC service.

This fails:
```python
c['services'].append(reporters.IRC("irc.oftc.net", "mybot",
                                   useColors=True,
                                   channels=[{"channel": "#test"}],
                                   password=util.Interpolate("%(secret:irc-pass)s"),
                                   notify_events={}))
```

This is buildbot 0.9.11 on Python 3.6.2

**Comment by tardyp** [comment]
This is partially implemented with https://github.com/buildbot/buildbot/pull/3357

With this PR, all we need now is to add a line to irc.py

```
class IRC(service.BuildbotService):
    secrets = ['password']
```

If you can test this patch and make sure it works for you, please submit a PR!

**Comment by berkerpeksag** [comment]
This feature has been implemented in https://github.com/buildbot/buildbot/commit/9e68ca502df2bfdb3b6eb5fb4ec9bb19b5e123c6. I think this issue can be closed now.

**Comment by tardyp** [comment]
@berkerpeksag the is another fix needed in 1.1.0 #3986
